core/src/main/java/org/infinispan/interceptors/impl/TxInterceptor.java

```diff
@@ -604,6 +604,8 @@ private Object verifyRemoteTransaction(RemoteTxInvocationContext ctx, AbstractTr
             log.tracef("Rolling back remote transaction %s because either already completed (%s) or originator no longer in the cluster (%s).",
                        globalTransaction, alreadyCompleted, originatorMissing);
          }
+         // The rollback command only marks the transaction as completed in invokeAsync()
+         txTable.markTransactionCompleted(globalTransaction, false);
```

Comment:
Is this the only place that marks transaction for rollback when the lock command fails? Or is it up to the user to handle exception being thrown by the operation and mark the tx?